### PR TITLE
dont leak the address string in UDPSocket.addressToString

### PR DESCRIPTION
**Pull request by nektro**

### CODE REVIEW

**Comment by robobun** [comment]
@nektro, your commit bf8f70436c73cf4f7890ea06bb3b14f7c4aaff5d is building: [`#3788`](https://buildkite.com/bun/bun/builds/3788)

File: src/bun.js/api/bun/udp_socket.zig [hunk 1]
```diff
@@ -631,7 +631,8 @@ pub const UDPSocket = struct {
         };
 
         const slice = bun.fmt.formatIp(address, &text_buf) catch unreachable;
-        return bun.String.createLatin1(slice).toJS(globalThis);
+        var str = bun.String.createLatin1(slice);
+        return str.transferToJS(globalThis);
```
Comment (cirospaciari) [inline src/bun.js/api/bun/udp_socket.zig:635]: nice finding

**Review by cirospaciari — CHANGES_REQUESTED** [review]
Can we add a memory leak test on UDP sockets that cover this case?

**Comment by Jarred-Sumner** [comment]
oh i pressed merge before I saw your comment

**Pull request merged**